modules/electromagnetics/include/kernels/ADMatWaveReaction.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADVectorKernel.h"
+
+/**
+ *  Calculates the current source term in the Helmholtz wave equation using
```

Comment:
you call it a source term here.
is Source better than Reaction?
if you are calling it Reaction because of the moose "it's a lambda u term" you can definitely go past that.
momentum friction sources/sinks in NS are not called reaction